pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
> On Apr 20, 2015, at 7:47 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > +	EndpointsName string `json:"endpoints"`
> > +	// Required: RBDImage is the rados image name
> > +	RBDImage string `json:"image"`
> > +	// Required: Filesystem type to mount.
> > +	// Must be a filesystem type supported by the host operating system.
> > +	// Ex. "ext4", "xfs", "ntfs"
> > +	// TODO: how do we prevent errors in the filesystem from compromising the machine
> > +	FSType string `json:"fsType,omitempty"`
> > +	// Optional: RBDPool is the rados pool name,default is rbd
> > +	RBDPool string `json:"pool"`
> > +	// Optional: RadosUser is the rados user name, default is admin
> > +	RadosUser string `json:"user"`
> > +	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
> > +	Keyring string `json:"keyring"`
> > +	// Optional: Secret is the authentication secret for RBDUser, default is empty.
> > +	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
> I'm so confused.
> 
> A claim (once satisfied) is just a symlink to a PersistentVolume. A PV holds PVS which holds RBDVolumeSource (this struct). We either
> 
> a) Make this struct hold an object ref and enforce the semantic we want
> b) Make two structs which are and must continue to be identical in every way, except SecretName vs SecretRef.
> 
> I say A. Tell me why this is better than forking the struct or why we don't need to fork the struct, cuz I am not seeing it.
> 
Why not one struct with both SecretName and secretref?  I'm not opposed to a single struct with object reference since reusing the struct is more efficient.  It's just a bit more documentation and explanation to users.

Consider objection to object ref and SecretRef withdrawn.
> I am fine with Endpoints as []string, but don't call it "Endpoints".
> 
From ceph docs this is usually called Host but can be a host:port.  Since hostports is overloaded, how about Servers?
> —
> Reply to this email directly or view it on GitHub.
>